cmd/aries-agent-mobile/pkg/wrappers/storage/storage.go

```diff
@@ -8,47 +8,366 @@ SPDX-License-Identifier: Apache-2.0
+	if err != nil {
+		// The errors returned from the mobile binding provider will just have the error text without any distinct
+		// error types like the ones defined in aries-framework-go/spi/storage. In order to still allow for the use of
+		// errors.Is(err, spi.ErrStoreNotFound) in higher level functions,
+		// we need to wrap spi.ErrStoreNotFound back in if we detect it.
+		if strings.Contains(err.Error(), spi.ErrStoreNotFound.Error()) {
+			return fmt.Errorf("failed to set store config in mobile binding provider: %s: %w", err.Error(),
+				spi.ErrStoreNotFound)
+		}
+
+		return fmt.Errorf("failed to set store config in mobile binding provider: %w", err)
+	}
+
+	return nil
 }
 
-type store struct{ api.Store }
+// GetStoreConfig gets the current store configuration.
+// The store must be created prior to calling this method.
+// If the store cannot be found, then an error wrapping ErrStoreNotFound will be returned.
+// If name is blank, then an error will be returned.
+func (p *Provider) GetStoreConfig(name string) (spi.StoreConfiguration, error) {
+	configBytes, err := p.mobileBindingProvider.GetStoreConfig(name)
+	if err != nil {
+		// The errors returned from the mobile binding provider will just have the error text without any distinct
+		// error types like the ones defined in aries-framework-go/spi/storage. In order to still allow for the use of
+		// errors.Is(err, spi.ErrStoreNotFound) in higher level functions,
+		// we need to wrap spi.ErrStoreNotFound back in if we detect it.
+		if strings.Contains(err.Error(), spi.ErrStoreNotFound.Error()) {
+			return spi.StoreConfiguration{},
+				fmt.Errorf("failed to get store config from mobile binding provider: %s: %w", err.Error(),
+					spi.ErrStoreNotFound)
+		}
+
+		return spi.StoreConfiguration{},
+			fmt.Errorf("failed to get store config from mobile binding provider: %w", err)
+	}
+
+	var storeConfig spi.StoreConfiguration
+
+	err = json.Unmarshal(configBytes, &storeConfig)
+	if err != nil {
+		return spi.StoreConfiguration{}, fmt.Errorf("failed to unmarshal store configuration bytes: %w", err)
+	}
+
+	return storeConfig, nil
+}
+
+// GetOpenStores returns all currently open stores.
+func (p *Provider) GetOpenStores() []spi.Store {
+	p.lock.RLock()
+	defer p.lock.RUnlock()
+
+	openSPIStores := make([]spi.Store, len(p.openStores))
+
+	var counter int
+
+	for _, openStoreWrapper := range p.openStores {
+		openSPIStores[counter] = openStoreWrapper
+		counter++
+	}
+
+	return openSPIStores
+}
+
+// Close closes all stores created under this store provider.
+// For persistent mobile binding store implementations, this does not delete any data in the underlying databases.
+func (p *Provider) Close() error {
+	p.lock.RLock()
+
+	openStoresSnapshot := make([]*store, len(p.openStores))
+
+	var counter int
+
+	for _, openStore := range p.openStores {
+		openStoresSnapshot[counter] = openStore
+		counter++
+	}
+	p.lock.RUnlock()
+
+	for _, openStore := range openStoresSnapshot {
```

Comment:
Don't you need a write lock here??

Reply:
I don't believe so. So each store implementation will have their own way of "closing" a store. For many, it's simply a no-op. For others, they may terminate a connection. If a store needs a write lock, they'll have to implement one inside their own methods...

Let me know if there's a specific scenario you can think of that would cause issues.